error.go

```diff
@@ -2,38 +2,42 @@
 // and efficient error handling.
 package microerror
 
-import (
-	"fmt"
-
-	"github.com/juju/errgo"
-)
-
 var (
-	// MaskAny is a simple error masker. Masked errors act as tracers within the
-	// source code. Inspecting an masked error shows where the error was passed
-	// through within the code base. This is gold for debugging and bug huntin.
-	MaskAny = errgo.MaskFunc(errgo.Any)
+	Default = NewErrgoHandler(ErrgoHandlerConfig{
+		CallDepth: 1,
```

Comment:
Why don't we use the default config here and why does the default config now set the call depth to 1? Something is still not right. I realized there are no tests for the default. See the traces showing `runtime/proc.go` references. This is wrong. 

```
xh3b4sd@osx.local go run main.go
[{/Users/xh3b4sd/go/src/github.com/giantswarm/test/main.go:15: } {/Users/xh3b4sd/go/src/github.com/giantswarm/test/main.go:14: } {/Users/xh3b4sd/go/src/github.com/giantswarm/test/main.go:13: } {test}]

[{/Users/xh3b4sd/go/src/github.com/giantswarm/test/main.go:27: 3} {/Users/xh3b4sd/go/src/github.com/giantswarm/test/main.go:26: 2} {/Users/xh3b4sd/go/src/github.com/giantswarm/test/main.go:25: 1} {test}]

[{/Users/xh3b4sd/.gvm/gos/go1.8.3/src/runtime/proc.go:185: } {/Users/xh3b4sd/.gvm/gos/go1.8.3/src/runtime/proc.go:185: } {/Users/xh3b4sd/.gvm/gos/go1.8.3/src/runtime/proc.go:185: } {test}]

[{/Users/xh3b4sd/.gvm/gos/go1.8.3/src/runtime/proc.go:185: 3} {/Users/xh3b4sd/.gvm/gos/go1.8.3/src/runtime/proc.go:185: 2} {/Users/xh3b4sd/.gvm/gos/go1.8.3/src/runtime/proc.go:185: 1} {test}]

xh3b4sd@osx.local cat main.go
package main

import (
	"fmt"

	"github.com/giantswarm/microerror"
)

func main() {
	{
		err := fmt.Errorf("test")

		err = microerror.Mask(err)
		err = microerror.Mask(err)
		err = microerror.Mask(err)

		fmt.Printf("%#v\n", err)
	}

	fmt.Printf("\n")

	{
		err := fmt.Errorf("test")

		err = microerror.Maskf(err, "1")
		err = microerror.Maskf(err, "2")
		err = microerror.Maskf(err, "3")

		fmt.Printf("%#v\n", err)

		fmt.Printf("\n")
	}

	{
		err := fmt.Errorf("test")

		err = microerror.Default.Mask(err)
		err = microerror.Default.Mask(err)
		err = microerror.Default.Mask(err)

		fmt.Printf("%#v\n", err)
	}

	fmt.Printf("\n")

	{
		err := fmt.Errorf("test")

		err = microerror.Default.Maskf(err, "1")
		err = microerror.Default.Maskf(err, "2")
		err = microerror.Default.Maskf(err, "3")

		fmt.Printf("%#v\n", err)

		fmt.Printf("\n")
	}
}
```